Task/2/teachers form

KuranovaPolina opened the pull request:
Added teacher info form without database

KuranovaPolina committed:
Added new files teacher_info.py, teacher_info_text.py, test_teacher_info_data.py

KuranovaPolina committed:
Added InlineKeyboardMarkup for teacher

KuranovaPolina committed:
Created teacher info form with "some [previous/next] teacher info" text

KuranovaPolina committed:
Added states with init.py and teacher_info_search_state.py; Added teacher's names polling with "echo" form

KuranovaPolina committed:
Unnecessary code deleted

KuranovaPolina committed:
Added form with first teacher or with "Преподаватель не найден" text

KuranovaPolina committed:
Added scroll to previous teacher

KuranovaPolina committed:
Added scroll to next teacher

KuranovaPolina committed:
teacher_info_text format edited

KuranovaPolina committed:
Added additional info check

callmemars1 commented on tg_bot/handlers/teacher_info.py:
```diff
@@ -0,0 +1,116 @@
+                                state=TeacherDialog.answer)
+
+
+async def ask_teacher(message: Message):
```
По названию функции кажется, будто мы о чем то спрашиваем учителя. Переименуй на что-то более подходящее

callmemars1 commented on tg_bot/handlers/teacher_info.py:
```diff
@@ -0,0 +1,116 @@
+from tg_bot.users import users, User
+
+
+async def display_teachers(message: Message, state: FSMContext):
```
Вынеси функции, отвечающий информацию о преподавателях в отдельный класс

callmemars1 commented on tg_bot/handlers/teacher_info.py:
```diff
@@ -0,0 +1,116 @@
+                             parse_mode="HTML")
+
+
+def register_display_teachers(dp: Dispatcher):
```
Регистрацию хендлеров тоже

KuranovaPolina committed:
Package states renamed into dialog_states

KuranovaPolina committed:
Unnecessary code deleted

KuranovaPolina committed:
ask_teacher and register_ask_teacher renamed into request_teacher_name and register_request_teacher_name

KuranovaPolina committed:
Teacher info functions and handlers registration placed in separate classes

KuranovaPolina committed:
Added service_handlers_registrars package
TeacherInfoServiceHandlersRegistrar.py renamed into teacher_info_service_handlers_registrar.py and placed in service_handlers_registrars

KuranovaPolina committed:
class TeacherInfo renamed into TeacherInfoService

KuranovaPolina committed:
Merge branch 'main' into task/2/teachers-form

callmemars1 commented on bot.py:
```diff
@@ -4,18 +4,20 @@
 from tg_bot.handlers.teacher_info import TeacherInfoService
 
+from TimetableMessageStatesMongoClient import TimetableMessageStatesMongoClient
+from TimetableServiceHandlersRegistrator import TimetableServiceHandlersRegistrator
```
Вот здесь тогда тоже на регистрар поменяй

KuranovaPolina committed:
TimetableServiceHandlersRegistrator.py renamed into timetable_service_handlers_registrar.py and moved into service_handlers_registrars

KuranovaPolina committed:
Added .idea into .gitignore

KuranovaPolina committed:
.idea, systemd deleted

KuranovaPolina committed:
All files returned to Git

KuranovaPolina committed:
All files returned to Git 2

KuranovaPolina committed:
systemd deleted from .gitignore

KuranovaPolina committed:
Unnecessary code deleted from .gitignore

callmemars1 commented on bot.py:
```diff
@@ -10,7 +10,7 @@
 
 from TimetableMessageStatesMongoClient import TimetableMessageStatesMongoClient
-from TimetableServiceHandlersRegistrator import TimetableServiceHandlersRegistrator
+from service_handlers_registrars.timetable_service_handlers_registrar import TimetableServiceHandlersRegistrator
```
Сам класс переименуй тоже

KuranovaPolina committed:
TimetableMessageStateDto.py renamed into timetable_message_state_dto.py
TimetableMessageStatesMongoClient.py renamed into timetable_message_states_mongo_client.py
TimetableServiceHandlerRegistrator renamed into TimetableServiceHandlerRegistrar

KuranovaPolina committed:
Merge branch 'main' into task/2/teachers-form

# Conflicts:
#	.idea/misc.xml
#	.idea/suai-bot-student-telegram-client.iml
#	bot.py
#	tg_bot/users.py
